### DISCUSSION TITLE
.net 6.0 compatibility

### DISCUSSION THREAD

**Discussion by greengumby** [wixtoolset/issues#6905] [Questions]
Hi,

Just noticed that its been over a year since Wix 4 preview.0 was released and If I read correctly even the preview.1 will not support .net 6.0.

So I have an application that is .NET 6.0 and if I use Wix 3.11 will the server require .NET framework 4.x just so I can install my product?

Thanks

**Comment by robmen** [comment]
The WiX toolset does not add any dependencies on any .NET version to your installation package.

The WiX toolset *does* require .NET on your _build server_ to create your installation packages. WiX v3 requires .NET Framework v3.5 to build your installation packages. WiX v4 requires .NET Framework v4.7.2 _or_ .NET Core v3.1 (the latter _may_ change to .NET 6.0 before WiX v4 GA) to build your installation packages.

**Reply by SamOxigen** [reply]
The problem with .net60.0 version is that handling customUI module through an assemblyInfo.cs is no longer supported in .net6.0.

**Comment by ssaarianm1** [comment]
This question was marked answered recently but the answer does not cover the concern and lock of feature I brought to  group attention.
If my question is not a concern then  "what is the alternative to AssemblyInfo.cs role for .Net 6.0 version?"

The answer "The WiX toolset does not add any dependencies on any .NET version to your installation package....." is true but is not helpful. You cannot GLUE the WPF dll to the wix. This is the problem.
Issues related to .Net6.0 is not only about dependencies that might exist  or not, it is also about feature on .Netframework 4.7+ that is no longet available in .Net6.0.

**Reply by robmen** [reply]
What is your scenario that requires .NET 6.0 to reference wix.dll v3.5?

**Reply by SamOxigen** [reply]
Earlier we developed a WPF custom UI (WPF .Netframework 4.7.2) and this was working OK with WiX Toolset v3.14.
Now we are upgrading our c# projects to .Net6.0 and the problem I have now is: how to let the wix know about the custom UI entry point, namely:

1.	In AssemblyInfo.cs file we had added following two lines:
using Microsoft.Tools.WindowsInstallerXml.Bootstrapper;
[assembly: BootstrapperApplication(typeof(InstallerSE.CustomUI))] 
2.	with BootstrapperCore.config file and with Wix bundle entry we were able to make our custom UI work with Wix:		
<BootstrapperApplicationRef Id="ManagedBootstrapperApplicationHost">
			
`<Payload SourceFile="$(var.InstallerSE.ProjectDir)\bin\$(var.Configuration)\BootstrapperCore.config"/>
			<Payload SourceFile="$(var.InstallerSE.ProjectDir)\bin\$(var.Configuration)\InstallerSE.dll"/>
			<Payload SourceFile="$(var.InstallerSE.ProjectDir)\bin\$(var.Configuration)\Jint.dll"/>
			<Payload SourceFile="$(var.InstallerSE.ProjectDir)\bin\$(var.Configuration)\Microsoft.Deployment.WindowsInstaller.dll"/>
			<Payload SourceFile="$(var.InstallerSE.ProjectDir)\bin\$(var.Configuration)\Newtonsoft.Json.dll" />`


Now that .Net6.0 doesn’t have an AssemblyInfo cs, I found no way to “glue” my custom UI to the Wix.

**Reply by robmen** [reply]
Ahh, you have a managed BootstrapperApplication that you want to use with .NET 6.0. That is supported in WiX v4.0-preview.1 (not before).

**Reply by SamOxigen** [reply]
let me try, thank you!

**Comment by robmen** [comment]
WiX v4.0-preview.1 will be the first release that supports building managed BootstrapperApplications using .NET 6.

(I guess I have to write this as a separate thread for it to be marked as an answer)l.

**Comment by Aditya-stack-bit** [comment]
Hi @robmen ,Even I am facing the same problem , Currently I have migrated one of my Application into .net 6 and the Wix tool set gives me the error  " **Error LGHT0103: The system cannot find the file '..\..\_Out\Release\App.exe**'.  so will you suggest that once I update WIX v4.0 then this will able to find particular exe.

**Reply by robmen** [reply]
That error means it can't find your file. You need to fix your code.
